frontend/src/services/CustomDropdownService.ts

```diff
@@ -1,217 +1,288 @@
-    }
-
-    const entityIds = selected.map(option => parseInt(option.id));
-    const patchObject = { [propertyToUpdate]: entityIds };
-
-    try {
-        const updateFunction = apiFunctionMap[entityType];
-        if (updateFunction) {
-            await updateFunction(entityId, patchObject);
-        } else {
-            console.error(`No update function found for entity type: ${entityType}`);
-        }
-    } catch (error) {
-        console.error(`Error updating ${entityType}`, error);
+export async function updateEntity({
+  selected,
+  entityId,
+  entityType,
+  propertyToUpdate,
+}: UpdateEntityParams) {
+  if (entityId == null) {
+    console.error(`Error updating ${entityType}`);
+    return;
+  }
+
+  const entityIds = selected.map((option) => parseInt(option.id));
+  const patchObject = { [propertyToUpdate]: entityIds };
+
+  try {
+    const updateFunction = apiFunctionMap[entityType];
+    if (updateFunction) {
+      await updateFunction(entityId, patchObject);
+    } else {
+      console.error(`No update function found for entity type: ${entityType}`);
     }
+  } catch (error) {
+    console.error(`Error updating ${entityType}`, error);
+  }
 }
 
-export function getConnectionId(formId: string, connections: ViaSerializerDetails[] | DestinationSerializerDetails[]): number | null {
-    const index = getFirstNumberFromString(formId)
-    if (index != null) {
-        return connections[index].id
-    }
-    return null
-
+export function getConnectionId(
+  formId: string,
+  connections: ViaSerializerDetails[] | DestinationSerializerDetails[],
+): number | null {
+  const index = getFirstNumberFromString(formId);
+  if (index != null) {
+    return connections[index].id;
+  }
+  return null;
 }
 
 function getFirstNumberFromString(inputString: string) {
-    const match = inputString.match(/\d+/);
-    return match ? parseInt(match[0], 10) : null;
+  const match = inputString.match(/\d+/);
+  return match ? parseInt(match[0], 10) : null;
 }
 
-export function searchFromEntitiesVia(searchValue: string, statement: ConnectivityStatement, formId: string): Option[] {
-    const viaIndex = getFirstNumberFromString(formId)
-    if(viaIndex == null || statement.vias == null || statement.vias[viaIndex] == null){
-        return []
-    }
-    const viaOrder = statement.vias[viaIndex].order
-    const anatomicalEntities = getEntitiesBeforeOrder(statement, viaOrder)
-
-    return mapAnatomicalEntitiesToOptions(searchAnatomicalEntities(anatomicalEntities, searchValue), 'From Entities')
+export function searchFromEntitiesVia(
+  searchValue: string,
+  statement: ConnectivityStatement,
+  formId: string,
+): Option[] {
+  const viaIndex = getFirstNumberFromString(formId);
+  if (
+    viaIndex == null ||
+    statement.vias == null ||
+    statement.vias[viaIndex] == null
+  ) {
+    return [];
+  }
+  const viaOrder = statement.vias[viaIndex].order;
+  const anatomicalEntities = getEntitiesBeforeOrder(statement, viaOrder);
```

Comment:
Although practically speaking it's unlikely that we will need it, we should still use the current selected options in the searchFromEntitiesX functions to populate the anatomicalEntities options (without creating duplicates)